service/src/main/java/uk/nhs/adaptors/gp2gp/common/storage/StorageConnectorService.java

```diff
@@ -0,0 +1,32 @@
+package uk.nhs.adaptors.gp2gp.common.storage;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import static uk.nhs.adaptors.gp2gp.gpc.GpcFileNameConstants.GPC_STRUCTURED_FILE_EXTENSION;
+
+import java.io.ByteArrayInputStream;
+
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.gp2gp.gpc.GpcStructuredResponseObject;
+
+@Service
+@AllArgsConstructor
+public class StorageConnectorService {
+    private final StorageConnector storageConnector;
+    private final ObjectMapper objectMapper;
+
+    @SneakyThrows(JsonProcessingException.class)
+    public void handleStructuredRecord(GpcStructuredResponseObject response) {
```

Comment:
I had mentioned in a comment below renaming `GpcStructuredResponseObject` to `StorageDataWrapper` since we can use this structure for all storage data. Then `handleStructuredRecord` could be renamed to uploadWithMetadata perhaps?